Controllers/StockController.cs

```diff
@@ -0,0 +1,282 @@
+            catch (Exception e)
+            {
+                return BadRequest($"Request failed. {e.Message}");
+            }
+        }
+
+        [HttpGet("{symbol}/pricetarget")]
+        public ActionResult PriceTarget(string symbol)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/stock/price-target?symbol={symbol.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                PriceTarget target = response.Content.ReadFromJsonAsync<PriceTarget>(serializerOptions).Result;
+
+                if (target is null)
+                {
+                    throw new Exception("Price target was empty.");
+                }
+
+                return Ok(target);
+            }
+            catch (Exception e)
+            {
+                return BadRequest($"Request failed. {e.Message}");
+            }
+        }
+
+        [HttpGet("{symbol}/quote")]
+        public ActionResult Quote(string symbol)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/quote?symbol={symbol.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                Quote quote = response.Content.ReadFromJsonAsync<Quote>(serializerOptions).Result;
+
+                if (quote is null)
+                {
+                    throw new Exception("Quote was empty.");
+                }
+
+                return Ok(quote);
+            }
+            catch (Exception e)
+            {
+                return BadRequest($"Request failed. {e.Message}");
+            }
+        }
+
+        [HttpGet("{symbol}/candles")]
+        public ActionResult Candles(string symbol, string resolution, DateTimeOffset from, DateTimeOffset to)
+        {
+            try
+            {
+                string parameters = $"https://finnhub.io/api/v1/stock/candle?symbol={symbol.ToUpperInvariant()}" +
+                                    $"&resolution={resolution}&from={from.ToUnixTimeSeconds()}&to={to.ToUnixTimeSeconds()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                Candles candles = response.Content.ReadFromJsonAsync<Candles>(serializerOptions).Result;
+
+                if (candles is null)
+                {
+                    throw new Exception("Candles are empty.");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/InvestingOak/InvestingOak/pullRequest?prid=6380270)